Tests/DatadogTests/Datadog/CrashReporting/CrashContext/CrashContextProviderTests.swift

```diff
@@ -14,355 +14,128 @@ import CoreTelephony
-        XCTAssertEqual(updatedContext?.lastUserInfo, randomUserInfo)
-    }
-
-    // MARK: - `NetworkConnectionInfo` Integration
-
-    func testWhenCurrentValueIsObtainedFromNetworkConnectionInfoProvider_thenCrashContextProviderNotifiesNewContext() throws {
-        let expectation = self.expectation(description: "Notify new crash context")
-        let initialNetworkConnectionInfo: NetworkConnectionInfo = .mockRandom()
-        let wrappedProvider = NetworkConnectionInfoProviderMock(networkConnectionInfo: initialNetworkConnectionInfo)
-        let mainProvider = NetworkConnectionInfoProvider(wrappedProvider: wrappedProvider)
-
-        let crashContextProvider = CrashContextProvider(
-            consentProvider: .mockAny(),
-            userInfoProvider: .mockAny(),
-            networkConnectionInfoProvider: mainProvider,
-            carrierInfoProvider: CarrierInfoProviderMock.mockAny(),
-            rumViewEventProvider: .mockRandom(),
-            rumSessionStateProvider: .mockAny(),
-            appStateListener: AppStateListenerMock.mockAny()
-        )
-
-        let initialContext = crashContextProvider.currentCrashContext
-        var updatedContext: CrashContext?
 
-        // When
-        crashContextProvider.onCrashContextChange = { newContext in
-            updatedContext = newContext
-            expectation.fulfill()
-        }
-        wrappedProvider.set(current: .mockRandom()) // change `NetworkConnectionInfo` in wrapped provider
-        let currentNetworkConnectionInfo = mainProvider.current // obtain new info through the main provider
+        core.send(message: .custom(key: "rum", baggage: ["view-event": viewEvent]))
+        core.send(message: .custom(key: "rum", baggage: ["view-reset": true]))
 
         // Then
-        waitForExpectations(timeout: 1, handler: nil)
-        let updatedNetworkConnectionInfo = try XCTUnwrap(updatedContext?.lastNetworkConnectionInfo)
-        XCTAssertEqual(initialContext.lastNetworkConnectionInfo, initialNetworkConnectionInfo, "It must store initial network info")
-        XCTAssertEqual(updatedNetworkConnectionInfo, currentNetworkConnectionInfo, "It must store updated network info")
+        waitForExpectations(timeout: 0.5, handler: nil)
+        XCTAssertNil(viewEvent)
     }
 
-    // MARK: - `CarrierInfo` Integration
-    #if !os(tvOS)
-
-    private let ctTelephonyNetworkInfoMock = CTTelephonyNetworkInfoMock(
-        serviceCurrentRadioAccessTechnology: ["000001": CTRadioAccessTechnologyLTE],
-        serviceSubscriberCellularProviders: ["000001": CTCarrierMock(carrierName: "Carrier", isoCountryCode: "US", allowsVOIP: true)]
-    )
+    // MARK: - RUM Session State Integration
 
-    func testGivenRunningOniOS11_whenCurrentValueIsObtainedFromCarrierInfoProvider_thenCrashContextProviderNotifiesNewContext() throws {
+    func testWhenRUMWithCrashContextIntegrationIsUpdatedWithRUMSessionState_thenCrashContextProviderNotifiesNewContext() {
```

Comment:
_"**when** new RUM new session state is sent through message bus, **then** it notifies new crash context_ 😉?